src/transformers/models/pop2piano/modeling_pop2piano.py

```diff
@@ -0,0 +1,1376 @@
+                https://huggingface.co/sweetcocoa/pop2piano/blob/main/generation_config.json .
+            generation_config (`~generation.GenerationConfig`, *optional*):
+                The generation configuration to be used as base parametrization for the generation call. `**kwargs`
+                passed to generate matching the attributes of `generation_config` will override them. If
+                `generation_config` is not provided, the default will be used, which had the following loading
+                priority: 1) from the `generation_config.json` model file, if it exists; 2) from the model
+                configuration. Please note that unspecified parameters will inherit [`~generation.GenerationConfig`]'s
+                default values, whose documentation should be checked to parameterize generation.
+            kwargs:
+                Ad hoc parametrization of `generate_config` and/or additional model-specific kwargs that will be
+                forwarded to the `forward` function of the model. If the model is an encoder-decoder model, encoder
+                specific kwargs should not be prefixed and decoder specific kwargs should be prefixed with *decoder_*.
+        Return:
+            [`~utils.ModelOutput`] or `torch.LongTensor`: A [`~utils.ModelOutput`] (if `return_dict_in_generate=True`
+            or when `config.return_dict_in_generate=True`) or a `torch.FloatTensor`.
+                Since Pop2Piano is an encoder-decoder model (`model.config.is_encoder_decoder=True`), the possible
+                [`~utils.ModelOutput`] types are:
+                    - [`~generation.GreedySearchEncoderDecoderOutput`],
+                    - [`~generation.SampleEncoderDecoderOutput`],
+                    - [`~generation.BeamSearchEncoderDecoderOutput`],
+                    - [`~generation.BeamSampleEncoderDecoderOutput`]
+        """
+
+        if generation_config is None:
+            generation_config = self.generation_config
+        generation_config.update(**kwargs)
+
+        # check for composer_to_feature_token
+        if not hasattr(generation_config, "composer_to_feature_token"):
+            raise ValueError(
+                "`composer_to_feature_token` was not found! Please refer to "
+                "https://huggingface.co/sweetcocoa/pop2piano/blob/main/generation_config.json"
+                "and parse a dict like that."
+            )
+        else:
+            if len(generation_config.composer_to_feature_token) != self.config.composer_vocab_size:
+                raise ValueError(
+                    "config.composer_vocab_size must be same as the number of keys in "
+                    f"generation_config.composer_to_feature_token! "
+                    f"Found {self.config.composer_vocab_size} vs {len(generation_config.composer_to_feature_token)}."
+                )
+
+        input_features, attention_mask = self.get_mel_conditioner_outputs(
```

Comment:
Still not fully sure why we only do this in generate and not the forward call - should this not be in the forward call?

Reply:
In general, there should be no additional modelling steps prior to the generation call. We might need to add this step to the T5Encoder so that calling `_prepare_encoder_decoder_kwargs_for_generation` computes the right ids in the first forward pass

Reply:
Actually `get_mel_conditioner_outputs` requires composer_token which we pass in the `generation_config` to generate different MIDI variants. Let me know if I should change this design or not.

Reply:
Okay good with me! Let's leave this thread here so the next reviewer knows :)

Reply:
done.